src/mesh/voxel_cloud.rs

```diff
@@ -448,6 +448,165 @@ impl ScalarField {
+                        Point3::from(corners_cartesian_coordinates[7].coords.lerp(&corners_cartesian_coordinates[4].coords, 0.5)), //e7
+                        Point3::from(corners_cartesian_coordinates[0].coords.lerp(&corners_cartesian_coordinates[4].coords, 0.5)), //e8
+                        Point3::from(corners_cartesian_coordinates[1].coords.lerp(&corners_cartesian_coordinates[5].coords, 0.5)), //e9
+                        Point3::from(corners_cartesian_coordinates[2].coords.lerp(&corners_cartesian_coordinates[6].coords, 0.5)), //e10
+                        Point3::from(corners_cartesian_coordinates[3].coords.lerp(&corners_cartesian_coordinates[7].coords, 0.5)), //e11
+                    ];
+
+                    // Boolean vector marking each of the voxels either volume
+                    // (`true`) or void(`false`)
+                    let corners_inside_volume_pattern: Vec<_> = corners_absolute_coordinates
+                        .iter()
+                        .map(|a| {
+                            self.value_at_absolute_voxel_coordinate(&a)
+                                .map(|value| volume_value_range.contains(&value))
+                                .unwrap_or(false)
+                        })
+                        .collect();
+
+                    // The marching cubes lookup table contains 256 possible
+                    // samples of mesh patches. The sample index specifies the
+                    // proper sample for the current voxel group.
+                    let mut sample_index = 0_u8;
+                    if corners_inside_volume_pattern[0] {
+                        sample_index |= 1;
+                    }
+                    if corners_inside_volume_pattern[1] {
+                        sample_index |= 2;
+                    }
+                    if corners_inside_volume_pattern[2] {
+                        sample_index |= 4;
+                    }
+                    if corners_inside_volume_pattern[3] {
+                        sample_index |= 8;
+                    }
+                    if corners_inside_volume_pattern[4] {
+                        sample_index |= 16;
+                    }
+                    if corners_inside_volume_pattern[5] {
+                        sample_index |= 32;
+                    }
+                    if corners_inside_volume_pattern[6] {
+                        sample_index |= 64;
+                    }
+                    if corners_inside_volume_pattern[7] {
+                        sample_index |= 128;
+                    }
+
+                    // The marching cubes lookup table sample, that contains
+                    // triplets of indices to the edge midpoints (currently
+                    // already vertices) that will become mesh patch faces.
+                    let marching_cubes_faces =
+                        marching_cubes_lookup_table[cast_usize(sample_index)].clone();
```

Comment:
The `.clone()` most likely won't be necessary once you incorporate my lookup table suggestion, but a slice would be enough in this case also - no need to have the whole vector if we just need a view.